src/Core/Content/Media/Infrastructure/Path/MediaUrlGenerator.php

```diff
@@ -27,6 +28,7 @@ public function generate(array $paths): array
                 $url = $value->path;
             } else {
                 $url = $this->filesystem->publicUrl($value->path);
```

Comment:
Have you tried just encoding `$value->path` before passing it into this method? That would be much simpler than doing it after the fact with all the url parsing and then only encoding the "path"